packages/typescript/src/index.js

```diff
@@ -58,6 +65,58 @@ module.exports = {
       },
     ],
 
+    // Recommended rules that require type information
+    '@typescript-eslint/no-unsafe-argument': 'off',
+    '@typescript-eslint/no-unsafe-assignment': 'off',
+    '@typescript-eslint/no-unsafe-call': 'off',
+    '@typescript-eslint/no-unsafe-member-access': 'off',
+    '@typescript-eslint/no-unsafe-return': 'off',
+    '@typescript-eslint/restrict-template-expressions': 'off',
+
+    // Our rules that require type information
+    '@typescript-eslint/consistent-type-exports': 'error',
+    '@typescript-eslint/naming-convention': [
+      'error',
+      {
+        selector: 'default',
+        format: ['camelCase'],
+        leadingUnderscore: 'forbid',
+        trailingUnderscore: 'forbid',
+      },
+      {
+        selector: 'enumMember',
+        format: ['PascalCase'],
+      },
+      {
+        selector: 'variable',
+        format: ['camelCase', 'UPPER_CASE', 'PascalCase'],
+      },
+      {
+        selector: 'typeLike',
+        format: ['PascalCase'],
+      },
+      {
+        selector: 'interface',
+        format: ['PascalCase'],
+        custom: {
+          regex: '^I[A-Z]',
+          match: false,
+        },
+      },
+    ],
+    '@typescript-eslint/no-confusing-void-expression': 'error',
```

Comment:
This is causing a lot of errors in `snaps-monorepo`, especially in functions that do something like `return fn()` where `fn` returns `void`. I think we should not enable this rule.

Reply:
Ah, that makes sense.

I can see how this rule might improve readability, but it would be more verbose, and might make it easier to forget to `return`.